src/get-started/first-java-service.md

```diff
@@ -0,0 +1,615 @@
+or is it redundant? -->
+
+The `MySchema` has a field `access` of type [`Prefixed`][prefixed], initialized in
+its constructor. It is a database access object, which allows to access
+the indexes of this service.
+
+[prefixed]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/storage/database/Prefixed.html
+
+To keep a registry of vehicles indexed by their IDs, we will use a `ProofMap`
+index with `String` keys and `Vehicle` values, named `vehicles`.
+The `ProofMap` ensures that the data is the same
+on each node in the network.
+We will expose our index through a factory method — a method that will create
+a new `ProofMap`. Use [`access.getProofMap`][access-get-proof-map] method
+to create the `vehicles` index:
+
+[access-get-proof-map]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/storage/database/Access.html#getProofMap(com.exonum.binding.core.storage.indices.IndexAddress,com.exonum.binding.common.serialization.Serializer,com.exonum.binding.common.serialization.Serializer)
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MySchema.java)
+inside_block:ci-vehicles
+<!--/codeinclude-->
+
+Notice that the `access.getProofMap` accepts three parameters:
+
+- an index _address_ identifying this index in the blockchain
+- two serializers: one for keys and one for values. Exonum needs the serializers
+  to convert objects into bytes and back, as it stores the objects as bytes.
+  For `String` keys, we use a standard serializer. For `Vehicle`s, which
+  are Protocol Buffers messages, we use a corresponding serializer for messages
+  of `Vehicle` type.
+
+!!! tip
+    As the [`StandardSerializers.protobuf`][protobuf-serializer] uses reflection
+    to look up the needed methods in the message class, it is recommended
+    to instantiate a protobuf serializer once for each type
+    _and_ keep it in a `static` field, e.g.:
+
+    ```java
+    private static final Serializer<Vehicle> VEHICLE_SERIALIZER =
+        StandardSerializers.protobuf(Vehicle.class);
+    ```
+
+[protobuf-serializer]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/common/serialization/StandardSerializers.html#protobuf(java.lang.Class)
+
+??? fail "Cannot import `Vehicle` class?"
+    In some IDEs, e.g. IntelliJ IDEA, it might be needed to manually
+    mark `car-registry-messages/target/generated-sources/protobuf/java`
+    directory as "Generated Sources Root" to be able to import the generated
+    classes:
+
+    1. Open context menu (right click on the directory)
+    2. "Mark Directory As" > "Generated Sources Root"
+
+Compile the project:
+
+```shell
+mvn compile
+```
+
+<!--
+todo: Shall we add a whole file of MySchema?
+-->
+
+<!--
+Todo: Shall we draw a parallel with the DAO objects in business
+applications?
+-->
+
+<!--
+Todo: Shall we include the references to extra info on MerkleDB,
```

Comment:
IMO, including references to the glossary or topical pages in the text makes sense  when the concept is mentioned the first time. (This is relevant not only for MerkleDB, but, say, for the service lifecycle later in the article.)

Reply:
References to definitions in the glossary worked well, thanks. You mention the service lifecycle — however, I intentionally did not describe it except of mentions of service artifacts and instances, because considered it out of scope. Do you think we shall mention it/link somehow?